### PR TITLE
added search contacts JSON request functionality and tests

### PR DESCRIPTION
**Pull request by ie208**
- Added a case to ServerMain to allow for the searchcontacts JSON request which will contain a search query that will search the database for usernames that match the query. The server returns a list of contacts, similar to getallcontacts JSON request.

-added a test case to the test project

### CODE REVIEW

**Comment by MetallicFocus** [comment]
Great! Whenever you've got time, please write on the wiki page the JSON request and response for `SEARCHCONTACTS`

**Comment by ie208** [comment]
The JSON request is already on the wiki. Ill add the reply now

**Comment by EmadNaguiDarwish** [comment]
Looks good. Will merge it now.

File: TeamZeroServer/src/server/DbConnection.java [hunk 1]
```diff
@@ -180,6 +180,40 @@ public Client authenticateUser(String userName, String password) {
...
+	/**
+	 * Gets a list of users from the database that match the query.
+	 * 
+	 * @return an arraylist of Client objects
+	 */
+	public ArrayList<Client> getSearchedUsers(String query) {
+		ArrayList<Client> searchedClients = new ArrayList<Client>();
+		Connection conn = this.connect();
+		try {
+			PreparedStatement ps = conn.prepareStatement("SELECT * FROM USERS WHERE lower(username) LIKE ?;");
```
Comment (MetallicFocus) [inline TeamZeroServer/src/server/DbConnection.java]: The query's parameter needs to be explicitly made lowercase too (i.e. lower(username) LIKE lower(?) ) to be case insensitive

**Pull request merged**